Carmunication/Controllers/MyPage/FriendAddViewController.swift

```diff
@@ -0,0 +1,153 @@
+        friendAddView.clearButton.addTarget(self, action: #selector(clearButtonPressed), for: .touchUpInside)
+        friendAddView.friendAddButton.addTarget(self, action: #selector(sendFriendRequest), for: .touchUpInside)
+
+        friendAddView.friendSearchTextField.delegate = self
+
+        let tapGesture = UITapGestureRecognizer(target: self, action: #selector(dismissTextField))
+        view.addGestureRecognizer(tapGesture)
+
+        // 키보드 노티피케이션 옵저버 등록
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillShow),
+            name: UIResponder.keyboardWillShowNotification,
+            object: nil
+        )
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(keyboardWillHide),
+            name: UIResponder.keyboardWillHideNotification,
+            object: nil
+        )
+    }
+
+    // 클래스의 인스턴스가 메모리에서 해제되기 전에 호출되는 메서드
+    deinit {
+        // 옵저버 해제
+        NotificationCenter.default.removeObserver(self, name: UIResponder.keyboardWillShowNotification, object: nil)
+        NotificationCenter.default.removeObserver(self, name: UIResponder.keyboardWillHideNotification, object: nil)
+    }
+
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        setButtonState(hasText: false)
+    }
+}
+
+// MARK: - @objc 메서드
+extension FriendAddViewController {
+
+    // 상단 닫기 버튼 클릭 시 동작
+    @objc private func closeFriendAddView() {
+        self.dismiss(animated: true)
+    }
+
+    // [검색] 버튼을 눌렀을 때 동작
+    @objc private func performFriendSearch() {
+        // TODO: - 친구 검색 기능 구현 필요
+        print("친구 검색")
+    }
+
+    // 텍스트필드 clear 버튼 눌렀을 때 동작
+    @objc func clearButtonPressed() {
+        friendAddView.friendSearchTextField.text = ""
+    }
+
+    // 텍스트 필드 비활성화 시 동작
+    @objc func dismissTextField() {
+        friendAddView.friendSearchTextFieldView.backgroundColor = .clear
+        friendAddView.friendSearchTextFieldView.layer.borderWidth = 1.0
+        friendAddView.friendSearchTextField.resignFirstResponder() // 최초 응답자 해제
+    }
+
+    // [친구 추가하기] 버튼 눌렀을 때 동작
+    @objc func sendFriendRequest() {
+        // TODO: - 친구 추가 구현 필요
+        print("친구 추가하기 버튼 클릭됨")
+    }
+
+    // 키보드가 나타날 때 호출되는 메서드
+    @objc func keyboardWillShow(notification: Notification) {
+        guard let userInfo = notification.userInfo else { return }
+        guard let keyboardFrame = (userInfo[UIResponder.keyboardFrameEndUserInfoKey] as? NSValue)?.cgRectValue else {
+            return
+        }
+
+        // 애니메이션을 사용하여 레이아웃 업데이트 → 친구 추가하기 버튼을 위로 올려준다.
+        UIView.animate(withDuration: 0.3) {
+            self.friendAddView.friendAddButton.transform = CGAffineTransform(
+                translationX: 0,
+                y: -keyboardFrame.height+24
+            )
+        }
+    }
+    // 키보드가 사라질 때 호출되는 메서드
```

Comment:
가독성을 위해 메서드간에 개행을 넣어주시면 좋을 것 같아요 😄

```suggestion
    }

    // 키보드가 사라질 때 호출되는 메서드
```

Reply:
코드에는 의도를 드러내는게 중요하다고 생각해요
모든 메서드 사이에 개행을 넣고 이 부분만 붙여놓으면 코드 읽다가 멈칫할 것 같거든요
근데, 여기에 시간과 에너지를 써야할만큼 강조해야하는 부분인가? 하면 그건 아닌거 같아요 🤔